internal/pkg/diagnostics/diagnostics.go

```diff
@@ -95,6 +95,23 @@ func GlobalHooks() Hooks {
 				return fileBytes
 			},
 		},
+		{
+			Name:        "environment",
+			Filename:    "environment.yml",
+			Description: "Environment variables",
+			ContentType: "application/yaml",
+			Hook: func(_ context.Context) []byte {
+				redacted, err := redactEnv()
+				if err != nil {
+					return []byte(err.Error())
+				}
+				out, err := yaml.Marshal(redacted)
```

Comment:
Are we guaranteed the content of the environment variables can always be serialized to YAML?

If that happens do we lose the whole environment file or only the one that failed to serialize?

I don't think this will be common but want to make sure we like the way it fails.